lib/internal/wrapAsync.js

```diff
@@ -0,0 +1,27 @@
+import identity from 'lodash/identity';
+import asyncify from '../asyncify';
+
+var supportsSymbol = typeof Symbol === 'function';
+
+function supportsAsync() {
+    var supported;
+    try {
+        /* eslint no-eval: 0 */
+        supported = isAsync(eval('(async function () {})'));
```

Comment:
`eval` can be abused in the same way that `new Function` can be abused. I'd rather just use `eval` here